nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.cpp

```diff
@@ -78,6 +96,11 @@ BT::NodeStatus RemoveInCollisionGoals::on_completion(
       "All goals are in collision and have been removed from the list");
   }
   setOutput("output_goals", valid_goal_poses);
+  // set `waypoint_statuses` output
+  if (waypoint_statuses_get_res) {
```

Comment:
You don't need the `if` here, it would just set it to an empty struct since `waypoint_statuses` was never populated.